rskj-core/src/main/java/org/ethereum/db/MutableRepository.java

```diff
@@ -406,4 +429,48 @@ private Optional<Keccak256> internalGetValueHash(byte[] key) {
         tracker.addNewReadKey(new ByteArrayWrapper(key));
         return mutableTrie.getValueHash(key);
     }
+
+    @Override
+    public void addTransientStorageRow(RskAddress addr, DataWord key, DataWord value) {
+        addTransientStorageBytes(addr, key, value.getByteArrayForStorage());
+    }
+
+    @Override
+    public void addTransientStorageBytes(RskAddress addr, DataWord key, byte[] value) {
+        byte[] triekey = trieKeyMapper.getAccountStorageKey(addr, key);
+
+        // Special case: if the value is an empty vector, we pass "null" which commands the trie to remove the item.
+        // Note that if the call comes from addStorageRow(), this method will already have replaced 0 by null, so the
+        // conversion here only applies if this is called directly. If suppose this only occurs in tests, but it can
+        // also occur in precompiled contracts that store data directly using this method.
+        if (value == null || value.length == 0) {
+            transientTrie.put(triekey, null);
+        } else {
+            transientTrie.put(triekey, value);
+        }
+    }
+
+    @Override
+    public void clearTransientStorage() {
+        this.transientTrie =  new MutableTrieImpl(new TrieStoreImpl(new HashMapDB()), new Trie());
```

Comment:
Basically after every transaction execution. The thing is that this class it's initialized everytime a transaction starts anyway. Because this is used for the Storage during the transaction execution that might be rolled back and used by the storage. This clear will be called at end of a transaction.